tdrs-backend/tdpservice/data_files/static/admin/js/admin/admin_datafile_model.js

```diff
@@ -0,0 +1,30 @@
+$(window).on('load', function() {
+    //your code here
+    console.log('loaded');
+    var submitBtn=document.querySelector('button[type=submit]');    // add the first listener
+    var theForm = submitBtn.parentNode.parentNode;
+
+    for (var i = 0; i < theForm.childNodes.length; i++) {
+        if (theForm.childNodes[i].className == "actions") {
+          form_header = theForm.childNodes[i];
+          break;
+        }        
+    }
+    for (var i = 0; i < form_header.childNodes.length; i++) {
+        if (form_header.childNodes[i].className == "action-counter") {
+          number_of_files = form_header.childNodes[i];
+          break;
+        }        
+    }
```

Comment:
```suggestion
    for (var i = 0; i < theForm.childNodes.length; i++) {
        if (theForm.childNodes[i].className == "actions") {
          form_header = theForm.childNodes[i];
         for (var k = 0; k < form_header.childNodes.length; k++) {
             if (form_header.childNodes[k].className == "action-counter") {
               number_of_files = form_header.childNodes[k];
               break;
        }        
    }
```

Any reason we couldn't nest loops like this? From my reading, we really only want to search the lists to find exactly one node under the form. I'm no JS native, but it feels like there might a simpler algorithm or function to find `action-counter`. What about things like `querySelector()` or `querySelectorAll()`?

Reply:
i left the bulk of the script the same, but did nest the loops like you suggested

Reply:
Do we know how big these node lists are? By nesting the loops we turned the algorithm from O(m + n) to O(m*n). If each list had 100 children thats 200 iterations versus 10,000 iterations.

Reply:
Oh I see the extra `break` is inside the `if` statement which keeps it O(m+n). I personally find the nesting of the loops confusing but I don't think you need to change it back since it works the same.

Reply:
@elipe17 this is looping over the html for admin page, I don't think we have more than 25-50 divs, etc as children.